components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/FileSecretRepository.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.mediation.security.vault;
+
+import org.wso2.securevault.secret.SecretRepository;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.wso2.securevault.DecryptionProvider;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Properties;
+
+/**
+ * Implementation of file based secret repository
+ */
+public class FileSecretRepository implements SecretRepository {
+
+    private static Log LOG = LogFactory.getLog(FileSecretRepository.class);
+
+    /* Parent secret repository */
+    private SecretRepository parentRepository;
+
+    @Override
+    public void init(Properties properties, String id) {
+        // nothing to do here
+    }
+
+    /**
+     * Returns the secret of provided alias name . An alias represents the logical name
+     * for a look up secret
+     *
+     * @param alias file path to the secret file
+     * @return
+     */
+    @Override
+    public String getSecret(String alias) {
+        // Read from file
+        String secretRawValue = getPlainTextSecret(alias);
+        DecryptionProvider decyptProvider = CipherInitializer.getInstance().getDecryptionProvider();
+
+        if (decyptProvider == null) {
```

Comment:
Let is throw an NPE. This needs to be detected at this level so that we won't miss it. Otherwise, there is a possibility that we miss the ERROR log and another error is thrown from somewhere else. Also it imposes the requirement of having to do a null check for the return string.